ContentType issue

# Bug report

Error occurred while running mypy with django-stubs's mypy plugin.

## What's wrong

#### Related code

```python
class Job(models.Model):
    uuid = models.UUIDField(db_index=True, default=uuid.uuid4, editable=False)
    type = models.CharField(max_length=50)  # noqa: A003
    start_date = models.DateTimeField(null=True)
    end_date = models.DateTimeField(null=True)
    unit_id = models.BigIntegerField(null=True)
    parent_job_uuid = models.UUIDField(null=True)

    updated_at = models.DateTimeField(auto_now=True)
    created_at = models.DateTimeField(auto_now_add=True)

    class Meta:
        db_table = 'job'

    @staticmethod
    def from_entity(job: entity.Job):
        return Job(
            uuid=job.uuid,
            type=job.job_type.value,
            start_date=job.start_date,
            end_date=job.end_date,
            unit_id=job.unit_id,
            parent_job_uuid=job.parent_job_uuid,
        )

```

#### Result of `mypy ./src`

```
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.8/3.8.3_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "mypy/checkexpr.py", line 3755, in accept
  File "mypy/checkexpr.py", line 263, in visit_call_expr
  File "mypy/checkexpr.py", line 340, in visit_call_expr_inner
  File "mypy/checkexpr.py", line 817, in check_call_expr_with_callee_type
  File "mypy/checkexpr.py", line 876, in check_call
  File "mypy/checkexpr.py", line 988, in check_callable_call
  File "mypy/checkexpr.py", line 716, in apply_function_plugin
  File "/Users/.../venv/lib/python3.8/site-packages/mypy_django_plugin/transformers/init_create.py", line 62, in redefine_and_typecheck_model_init
    return typecheck_model_method(ctx, django_context, model_cls, '__init__')
  File "/Users/.../venv/lib/python3.8/site-packages/mypy_django_plugin/transformers/init_create.py", line 36, in typecheck_model_method
    expected_types = django_context.get_expected_types(typechecker_api, model_cls, method=method)
  File "/Users/.../venv/lib/python3.8/site-packages/mypy_django_plugin/django/context.py", line 154, in get_expected_types
    from django.contrib.contenttypes.fields import GenericForeignKey
  File "/Users/.../venv/lib/python3.8/site-packages/django/contrib/contenttypes/fields.py", line 6, in <module>
    from django.contrib.contenttypes.models import ContentType
  File "/Users/.../venv/lib/python3.8/site-packages/django/contrib/contenttypes/models.py", line 133, in <module>
    class ContentType(models.Model):
  File "/Users/.../venv/lib/python3.8/site-packages/django/db/models/base.py", line 112, in __new__
    raise RuntimeError(
RuntimeError: Model class django.contrib.contenttypes.models.ContentType doesn't declare an explicit app_label and isn't in an application in INSTALLED_APPS.
.../models.py:17: error: INTERNAL ERROR -- Please try using mypy master on Github:
```

FYI. Line `17` is `return Job(`

#### `mypy.ini`

```ini
[mypy]
python_version = 3.8

##

allow_redefinition = True

# Plugins

plugins = pydantic.mypy,mypy_django_plugin.main

[mypy.plugins.django-stubs]
django_settings_module = 'statsdjango.stats_django.settings'

[mypy-*.tests.*,tests.*]
ignore_errors = True

[mypy-*.migrations.*]
ignore_errors = True
```

---

Weird thing is... if I add to `INSTALLED_APPS` of my project's `settings.py` this error is gone.

## How is that should be

No error.

## System information

- OS: macOS Catalina
- `python` version: 3.8
- `django` version: 3.0.6
- `mypy` version: 0.782
- `django-stubs` version: master (https://github.com/typeddjango/django-stubs/commit/97ec2ee43b17bad55c68994c917321f5abdd1a0e)

--- a/mypy_django_plugin/django/context.py
+++ b/mypy_django_plugin/django/context.py
@@ -148,7 +148,9 @@ class DjangoContext:
         raise ValueError("No primary key defined")
 
     def get_expected_types(self, api: TypeChecker, model_cls: Type[Model], *, method: str) -> Dict[str, MypyType]:
-        from django.contrib.contenttypes.fields import GenericForeignKey
+        contenttypes_in_apps = self.apps_registry.is_installed("django.contrib.contenttypes")
+        if contenttypes_in_apps:
+            from django.contrib.contenttypes.fields import GenericForeignKey
 
         expected_types = {}
         # add pk if not abstract=True
@@ -192,7 +194,7 @@ class DjangoContext:
 
                     expected_types[field_name] = model_set_type
 
-            elif isinstance(field, GenericForeignKey):
+            elif contenttypes_in_apps and isinstance(field, GenericForeignKey):
                 # it's generic, so cannot set specific model
                 field_name = field.name
                 gfk_info = helpers.lookup_class_typeinfo(api, field.__class__)

--- a/mypy_django_plugin/transformers/request.py
+++ b/mypy_django_plugin/transformers/request.py
@@ -8,6 +8,9 @@ from mypy_django_plugin.lib import helpers
 
 
 def set_auth_user_model_as_type_for_request_user(ctx: AttributeContext, django_context: DjangoContext) -> MypyType:
+    if not django_context.apps_registry.is_installed("django.contrib.auth"):
+        return ctx.default_attr_type
+
     # Imported here because django isn't properly loaded yet when module is loaded
     from django.contrib.auth.base_user import AbstractBaseUser
     from django.contrib.auth.models import AnonymousUser

--- a/tests/typecheck/test_request.yml
+++ b/tests/typecheck/test_request.yml
@@ -27,6 +27,14 @@
             reveal_type(request.user) # N: Revealed type is 'django.contrib.auth.models.User'
     custom_settings: |
         INSTALLED_APPS = ('django.contrib.contenttypes', 'django.contrib.auth')
+-   case: request_object_user_without_auth_and_contenttypes_apps
+    disable_cache: true
+    main: |
+        from django.http.request import HttpRequest
+        request = HttpRequest()
+        reveal_type(request.user) # N: Revealed type is 'Union[django.contrib.auth.base_user.AbstractBaseUser, django.contrib.auth.models.AnonymousUser]'
+        if request.user.is_authenticated:
+            reveal_type(request.user) # N: Revealed type is 'django.contrib.auth.base_user.AbstractBaseUser'
 -   case: subclass_request_not_changed_user_type
     disable_cache: true
     main: |

--- a/tests/typecheck/test_request.yml
+++ b/tests/typecheck/test_request.yml
@@ -35,6 +35,16 @@
         reveal_type(request.user) # N: Revealed type is 'Union[django.contrib.auth.base_user.AbstractBaseUser, django.contrib.auth.models.AnonymousUser]'
         if request.user.is_authenticated:
             reveal_type(request.user) # N: Revealed type is 'django.contrib.auth.base_user.AbstractBaseUser'
+-   case: request_object_user_without_auth_but_with_contenttypes_apps
+    disable_cache: true
+    main: |
+        from django.http.request import HttpRequest
+        request = HttpRequest()
+        reveal_type(request.user) # N: Revealed type is 'Union[django.contrib.auth.base_user.AbstractBaseUser, django.contrib.auth.models.AnonymousUser]'
+        if request.user.is_authenticated:
+            reveal_type(request.user) # N: Revealed type is 'django.contrib.auth.base_user.AbstractBaseUser'
+    custom_settings: |
+        INSTALLED_APPS = ('django.contrib.contenttypes',)
 -   case: subclass_request_not_changed_user_type
     disable_cache: true
     main: |

